flask-app/app.py

```diff
@@ -0,0 +1,186 @@
+  crowd = data['crowd']
+  gameplay = data['gameplay']
+
+  query = """ UPDATE labelled
+              SET
+                commentator = {:},
+                crowd = {:},
+                gameplay = {:},
+                labelled = 1,
+                updated_at = NOW()
+              WHERE url = '{:}'
+              AND start = {:}
+              AND end = {:} """.format(commentator, crowd, gameplay, url, start, end)
+  return query
+
+
+def formatFormInput(formData):
+  try:
+    if formData['shortcut']:
+      inputText = formData['shortcut'].split(',')
+      if len(inputText) < 3:
+        return {'error': 'missing values'}
+
+      data = {
+        'commentator': int(inputText[0].strip()) - 1,
+        'crowd': int(inputText[1].strip()) - 1,
+        'gameplay': int(inputText[2].strip()) - 1,
+      }
+    else:
+      if 'commentator' not in formData or 'crowd' not in formData or 'gameplay' not in formData:
+        return {'error': 'missing values'}
+
+      data = {
+        'commentator': int(formData['commentator']),
+        'crowd': int(formData['crowd']),
+        'gameplay': int(formData['gameplay']),
+      }
+    return data
+  except Exception as e:
+    return {'error': e}
+
+
+def validateFormInput(data):
+  if 'error' in data:
+    return data
+  
+  if not (0 <= data['commentator'] <= 2):
+    data['error'] = 'commentator excitement val must be btwn 1 and 3'
+    return data
+
+  if not (0 <= data['crowd'] <= 2):
+    data['error'] = 'crowd excitement val must be btwn 1 and 3'
+    return data
+  
+  if not (0 <= data['gameplay'] <= 1):
+    data['error'] = 'gameplay val must be btwn 1 and 2'
+    return data
+  
+  return data
+
+
+@app.route("/addLabel", methods=['POST'])
+def addLabel():
+  try:
+    url = request.form['youtubeUrl']
+    start = request.form['start']
+    end = request.form['end']
+
+    data = formatFormInput(request.form)
+    validatedData = validateFormInput(data)
+    # re-render with error msg if error
+    if 'error' in validatedData:
+      error = validatedData['error']
+      embed_url = formatYoutubeEmbedUrl(url, start, end)
+      return render_template('index.html', embedUrl= embed_url, youtubeUrl= url, start= start, end= end, error= error)
+    
+    # connect to db 
+    conn = mysql.connect()
+    cur = conn.cursor()
+    # call update query
+    query = formatAddLabelQuery(url, start, end, data)
+    cur.execute(query)
+    conn.commit()
+    # close connection
+    cur.close()
+    conn.close()
+
+  except Exception as e:
+    return json.dumps({'error': str(e)})
+  
+  return redirect(url_for('index')) #redirecting to home
+
+
+def formatInsertQuery(data):
+  rows = []
+  for clip in data:
+    url = clip['youtubeUrl']
+    start = clip['start']
+    end = clip['end']
+    rows.append("('{:}', {:} , {:})".format(url, start, end))
+  
+  query = "INSERT INTO labelled(url, start, end) VALUE {:}".format(",".join(rows))
+  return query
+
+
+@app.route("/addClips", methods=['POST'])
```

Comment:
seems a bit exotic to do the db insert from the flask app? For the convenience of the pymysql libs we are using? Exotic - but no showstoppers <3

Reply:
Ye I agree, def not the most secure thing to do. I was lazy and didn't want to create sql connections on the node project as well.